### PR TITLE
Remove EBUS macros from GEMS

### PR DESCRIPTION
**Pull request by amzn-ahmadkrm**
### **What does this PR do?**

Replaces more EBUS_EVENT style macros. This is following on from the changes made by @greerdv in https://github.com/o3de/o3de/pull/14404.

### **How was this PR tested?**

Built and ran tests.

### CODE REVIEW

**Review by moraaar — APPROVED** [review]
Thanks for replacing the old macros!

**Review by rgba16f — APPROVED** [review]
Atom changes look good.

File: Gems/EMotionFX/Code/Source/Editor/PropertyWidgets/ActorJointHandler.cpp [hunk 1]
```diff
@@ -225,7 +225,7 @@ namespace EMotionFX
 
         connect(picker, &ActorJointPicker::SelectionChanged, this, [picker]()
         {
-            EBUS_EVENT(AzToolsFramework::PropertyEditorGUIMessages::Bus, RequestWrite, picker);
+            AzToolsFramework::PropertyEditorGUIMessages::Bus::Broadcast(&AzToolsFramework::PropertyEditorGUIMessages::Bus::Events::RequestWrite, picker);
             AzToolsFramework::PropertyEditorGUIMessages::Bus::Broadcast(&AzToolsFramework::PropertyEditorGUIMessages::Bus::Handler::OnEditingFinished, picker);
```
Comment (lumberyard-employee-dm) [inline Gems/EMotionFX/Code/Source/Editor/PropertyWidgets/ActorJointHandler.cpp:229]: These EBus calls can be grouped together to form a single call.
```suggestion
            AzToolsFramework::PropertyEditorGUIMessages::Bus::Broadcast([](AzToolsFramework::PropertyEditorGUIMessages* handler) {
            handler->RequestWrite(picker);
            handler->OnEditingFinished(picker);
            });
```

File: Gems/VideoPlaybackFramework/Code/Source/VideoPlaybackFrameworkSystemComponent.cpp [hunk 2]
```diff
@@ -137,11 +137,11 @@ namespace VideoPlaybackFramework
...
-        EBUS_EVENT(AZ::Data::AssetCatalogRequestBus, EnableCatalogForAsset, azrtti_typeid<VideoPlaybackAsset>());
-        EBUS_EVENT(AZ::Data::AssetCatalogRequestBus, AddExtension, "mp4");
-        EBUS_EVENT(AZ::Data::AssetCatalogRequestBus, AddExtension, "mkv");
-        EBUS_EVENT(AZ::Data::AssetCatalogRequestBus, AddExtension, "webm");
-        EBUS_EVENT(AZ::Data::AssetCatalogRequestBus, AddExtension, "mov");
+        AZ::Data::AssetCatalogRequestBus::Broadcast(&AZ::Data::AssetCatalogRequestBus::Events::EnableCatalogForAsset, azrtti_typeid<VideoPlaybackAsset>());
+        AZ::Data::AssetCatalogRequestBus::Broadcast(&AZ::Data::AssetCatalogRequestBus::Events::AddExtension, "mp4");
+        AZ::Data::AssetCatalogRequestBus::Broadcast(&AZ::Data::AssetCatalogRequestBus::Events::AddExtension, "mkv");
+        AZ::Data::AssetCatalogRequestBus::Broadcast(&AZ::Data::AssetCatalogRequestBus::Events::AddExtension, "webm");
+        AZ::Data::AssetCatalogRequestBus::Broadcast(&AZ::Data::AssetCatalogRequestBus::Events::AddExtension, "mov");
```
Comment (lumberyard-employee-dm) [inline Gems/VideoPlaybackFramework/Code/Source/VideoPlaybackFrameworkSystemComponent.cpp:144]: As the AssetCatalogRequestBus has a lock on it, each of these 5 calls have to take a mutex each time.
This is a good candidate to combine into a single call using a lambda.
```suggestion
        AZ::Data::AssetCatalogRequestBus::Broadcast([](AssetCatalogRequests* handler){
        handler->EnableCatalogForAsset(azrtti_typeid<VideoPlaybackAsset>());
        handler->AddExtension("mp4");
        handler->AddExtension("mkv");
        handler->AddExtension("webm");
        handler->AddExtension("move");
        });
```

**Review by lumberyard-employee-dm — APPROVED** [review]

**Comment by hultonha** [comment]
Temporarily moving to Draft while the author is away - will merge on their return

**Review by hultonha — APPROVED** [review]
Nice work @amzn-ahmadkrm 😄

**Pull request merged**